pkg/controller/backup_configuration.go

```diff
@@ -0,0 +1,356 @@
+	"k8s.io/client-go/tools/reference"
+)
+
+func (c *StashController) initBackupConfigurationWatcher() {
+	c.bupcInformer = c.stashInformerFactory.Stash().V1beta1().BackupConfigurations().Informer()
+	c.bupcQueue = queue.New("BackupConfiguration", c.MaxNumRequeues, c.NumThreads, c.runBackupConfigurationInjector)
+	c.bupcInformer.AddEventHandler(&cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			if b, ok := obj.(*api_v1beta1.BackupConfiguration); ok {
+				ref, rerr := reference.GetReference(scheme.Scheme, b)
+				if rerr == nil {
+					c.recorder.Eventf(
+						ref,
+						core.EventTypeWarning,
+						eventer.EventReasonInvalidBackupConfiguration,
+						"Reason: %v",
+						//err,
+					)
+				}
+				//return
+			}
+
+			queue.Enqueue(c.bupcQueue.GetQueue(), obj)
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			oldBupc, ok := oldObj.(*api_v1beta1.BackupConfiguration)
+			if !ok {
+				log.Errorln("Invalid BackupConfiguration object")
+				return
+			}
+			newbupc, ok := newObj.(*api_v1beta1.BackupConfiguration)
+			if !ok {
+				log.Errorln("Invalid BackupConfiguration object")
+				return
+			}
+			ref, rerr := reference.GetReference(scheme.Scheme, newbupc)
+			if rerr == nil {
+				c.recorder.Eventf(
+					ref,
+					core.EventTypeWarning,
+					eventer.EventReasonInvalidBackupConfiguration,
+					"Reason: %v",
+					//err,
+				)
+			}
+			if !util.BackupConfigurationEqual(oldBupc, newbupc) {
+				queue.Enqueue(c.bupcQueue.GetQueue(), newObj)
+			}
+
+		},
+		DeleteFunc: func(obj interface{}) {
+			queue.Enqueue(c.bupcQueue.GetQueue(), obj)
+
+		},
+	})
+	c.bupcLister = c.stashInformerFactory.Stash().V1beta1().BackupConfigurations().Lister()
+}
+
+// syncToStdout is the business logic of the controller. In this controller it simply prints
+// information about the deployment to stdout. In case an error happened, it has to simply return the error.
+// The retry logic should not be part of the business logic.
+func (c *StashController) runBackupConfigurationInjector(key string) error {
+	obj, exists, err := c.bupcInformer.GetIndexer().GetByKey(key)
+	if err != nil {
+		glog.Errorf("Fetching object with key %s from store failed with %v", key, err)
+		return err
+	}
+	if !exists {
+		glog.Errorf("BackupConfiguration %s does not exit anymore\n", key)
+		namespace, name, err := cache.SplitMetaNamespaceKey(key)
+		if err != nil {
+			return err
+		}
+
+		c.EnsureSidecarDeleted2(namespace, name)
+		err = c.EnsureCronJobDeleted(namespace, name)
+		fmt.Println(err)
+		if err != nil {
+			return err
+		}
+	} else {
+		backupconfiguration := obj.(*api_v1beta1.BackupConfiguration)
+		glog.Info("Sync/Add/Update for BackupConfiguration %s", backupconfiguration.GetName())
+		if backupconfiguration.Spec.Target != nil {
+			if backupconfiguration.Spec.Target.Ref != nil {
+				kind := backupconfiguration.Spec.Target.Ref.Kind
+				if util.BackupModel(kind) == util.ModelSidecar {
+					c.EnsureSidecar2(backupconfiguration)
```

Comment:
Rename `EnsureSidecar2` to `EnsureV1beta1Sidecar`

Reply:
fixed